services/web/server/src/simcore_service_webserver/scicrunch/service_client.py

```diff
@@ -0,0 +1,189 @@
+from .scicrunch_models import ListOfResourceHits, ResourceView
+
+logger = logging.getLogger(__name__)
+
+
+async def get_all_versions(
+    rrid: str, client: ClientSession, settings: SciCrunchSettings
+) -> List:
+    async with client.get(
+        f"{settings.api_base_url}/resource/versions/all/{rrid}",
+        params={"key": settings.api_key.get_secret_value()},
+        raise_for_status=True,
+    ) as resp:
+        body = await resp.json()
+        return body.get("data") if body.get("success") else []
+
+
+async def get_resource_fields(
+    rrid: str, client: ClientSession, settings: SciCrunchSettings
+) -> ResourceView:
+    async with client.get(
+        f"{settings.api_base_url}/resource/fields/view/{rrid}",
+        params={"key": settings.api_key.get_secret_value()},
+        raise_for_status=True,
+    ) as resp:
+        body = await resp.json()
+
+        assert body.get("success")  # nosec
+        return ResourceView(**body.get("data", {}))
+
+
+async def autocomplete_by_name(
+    guess_name: str, client: ClientSession, settings: SciCrunchSettings
+) -> ListOfResourceHits:
+    async with client.get(
+        f"{settings.api_base_url}/resource/fields/autocomplete",
+        params={
+            "key": settings.api_key.get_secret_value(),
+            "field": "Resource Name",
+            "value": guess_name.strip(),
+        },
+        raise_for_status=True,
+    ) as resp:
+        body = await resp.json()
+        assert body.get("success")  # nosec
+        return ListOfResourceHits.parse_obj(body.get("data", []))
+
+
+class ValidationResult(IntEnum):
+    UNKNOWN = -1
+    INVALID = 0
+    VALID = 1
+
+
+class SciCrunchAPI:
+    """Client to communicate with scicrunch.org service
+
+    - wraps all calls to scicrunch.org API
+        - return result or raises web.HTTPError
+    - one instance per application
+        - uses app aiohttp client session instance
+        - uses settings
+    """
+
+    # FIXME: scicrunch timeouts should raise -> Service not avialable with a msg that scicrunch is currently not responding
+
+    RRID_RE = re.compile(RRID_PATTERN)
+
+    def __init__(self, client: ClientSession, settings: SciCrunchSettings):
+        self.settings = settings
+        self.client = client
+        self.base_url = URL.build(
+            scheme=self.settings.api_base_url.scheme,
+            host=self.settings.api_base_url.host,
+        )
+
+    # Website links ---------
+
+    def get_portal_link(self) -> str:
+        return str(self.base_url.with_path("/resources/"))
+
+    def get_rrid_link(self, rrid: str) -> str:
+        # NOTE: for some reason scicrunch query does not like prefix!
+        return str(
+            self.base_url.with_path("/resources/Any/search").with_query(
+                q="undefined", l=rrid.replace("RRID: ", "").strip()
+            )
+        )
+
+    def get_new_submission_link(self) -> str:
+        return str(self.base_url.with_path("/resources/about/resource"))
```

Comment:
I would also move this in the `__init__` block

Reply:
done in PR #2065